corehq/apps/hqwebapp/static/hqwebapp/js/knockout_bindings.ko.js

```diff
@@ -260,19 +260,24 @@ hqDefine("hqwebapp/js/knockout_bindings.ko", ['jquery', 'knockout', 'jquery-ui/u
                 $(this).parent().parent().addClass("moving").siblings().removeClass('moving');
 
                 var row = $(this).parent().parent();
-                var currentIndex = row[0].attributes['data-order'].value;
-
-                var lastSelectedRowIndex = null;
-                for (var i = 0; i < list().length; i++) {
-                    if (list()[i].selected()) {
-                        lastSelectedRowIndex = i;
+                var currentIndex = parseInt(row[0].attributes['data-order'].value);
+                if (list()[currentIndex].selected()) {
+                    var lastSelectedRowIndex = null;
+                    for (var i = 0; i < list().length; i++) {
+                        if (list()[i].selected()) {
+                            lastSelectedRowIndex = i;
+                        }
                     }
-                }
 
-                if (currentIndex < lastSelectedRowIndex) {
+                    if (currentIndex < lastSelectedRowIndex) {
+                        var currentListItem = list.splice(currentIndex, 1)[0];
+                        row.remove();
+                        list.splice(lastSelectedRowIndex, 0, currentListItem);
```

Comment:
nit: this function is now really long, and has duplicated functionality. Do you think you could break some of this logic out into named functions so it is easier to read? Like this whole if statement could be called `sendToBottom` or something.